src/common/opencl.c

```diff
@@ -400,7 +400,7 @@ static int dt_opencl_device_init(dt_opencl_t *cl, const int dev, cl_device_id *d
   escapedkerneldir = dt_util_str_replace(kerneldir, " ", "\\ ");
 #endif
 
-  options = g_strdup_printf("-cl-mad-enable %s -D%s=1 -I%s",
+  options = g_strdup_printf("-w -Werror -cl-std=CL1.2 -cl-opt-disable %s -D%s=1 -I%s",
```

Comment:
We should do that for Intel AMD but not Nvidia since it works there. I don't this we want Nvidia users to be imposed a huge slowdown if the bug is not there.

Reply:
`if(!strncasecmp(vendor, "NVIDIA", 6))`

Reply:
Even for Intel & AMD users, that's something I would prefer to avoid since the perf penalty is huge and everything else in dt seems to be working correctly. I will keep looking for the exact origin of the problem, and if I have no success in a couple of weeks, then we can fallback to this ugly hack.

Reply:
Yes, I understand...

Reply:
These options don't work for AMD ROCm. It doesn't understand `-cl-opt-disable`.

Reply:
I would love it if vendors started to support **the full set of standard instructions**. :man_facepalming: